### PR TITLE
Extend tested Python versions

### PR DESCRIPTION
**Pull request by PeterJCLaw**
~Builds on https://github.com/danpalmer/django-devdata/pull/8 to pick up the CI fixes which that includes.~

It looks like Django 4 will need some actual code changes, or at least test setup changes. This PR therefore only looks at Python versions.

Also included here is a fix such that the Django "3.0" tests actually only test `3.0` and not everything.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,8 +12,22 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
-        django-version: [2.2, 3.0, 3.1, 3.2]
+        python-version: [3.7, 3.8, 3.9, "3.10"]
+        django-version: [2.2, 3.0, 3.1, 3.2, 4.0, 4.1]
+        exclude:
+          - python-version: "3.10"
```
Comment (danpalmer) [inline .github/workflows/ci.yml:19]: Minor: a comment at the top of each block explaining the motivation would be nice. Just something like:

> Django < 4 requires Python < 3.10

Also, unless you can find a nicer way to express that, maybe worth swapping the python/django keys so that they read like the comment.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -66,15 +80,15 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
 
-      - name: Set up Python ${{ matrix.python-version }}
```
Comment (danpalmer) [inline .github/workflows/ci.yml:69]: Was this working at all? If so, do you know where the matrix was coming from?

File: tests/testsite/testsite/settings/django.py [hunk 3]
```diff
@@ -67,6 +67,8 @@
     }
 }
 
+DEFAULT_AUTO_FIELD = "django.db.models.AutoField"
```
Comment (danpalmer) [inline tests/testsite/testsite/settings/django.py]: Is this new? Looks unrelated to the test changes?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (PeterJCLaw) [reply .github/workflows/ci.yml:19]: Yup, can definitely add a comment here.

In terms of ordering, I had been ordering it by Django version, though that was somewhat accidental. Happy to re-order these.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (PeterJCLaw) [reply .github/workflows/ci.yml:69]: Based on https://github.com/danpalmer/django-devdata/runs/7772449153?check_suite_focus=true (from the no-op PR #10) it was evaluating to an empty string.

File: tests/testsite/testsite/settings/django.py (hunk 3 above)
Comment (PeterJCLaw) [reply tests/testsite/testsite/settings/django.py]: This is new; I believe this setting is required in Django 4.

**Comment by PeterJCLaw** [comment]
I think the remaining test failures are somewhat legit -- I can reproduce the issue on Django 4+ but not 3.2 or below. I'm guessing something has changed in how migrations data are stored (at least in tests) in Django 4, though from a quick look through the release notes I didn't see anything.

I might unpick this to separate out the Django 4+ changes from the other changes as I suspect the rest are useful enough as they are.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (PeterJCLaw) [reply .github/workflows/ci.yml:19]: Included in a rebased version of the commit: 3038e6a9dec5c82218ad69b75ecc3a7f52a12424

File: tests/testsite/testsite/settings/django.py (hunk 3 above)
Comment (PeterJCLaw) [reply tests/testsite/testsite/settings/django.py]: Have rebased & dropped the Django 4 stuff; this disappears for now.

**Review by danpalmer — APPROVED** [review]

**Comment by danpalmer** [comment]
Thanks @PeterJCLaw!
